ethcore/sync/src/chain/supplier.rs

```diff
@@ -145,14 +145,47 @@ impl SyncSupplier {
 				}
 			};
 
-			result.unwrap_or_else(|e| {
-				debug!(target:"sync", "{} -> Malformed packet {} : {}", peer, packet_id, e);
-			})
+			match result {
+				Err(PacketProcessError::Decoder(e)) => debug!(target:"sync", "{} -> Malformed packet {} : {}", peer, packet_id, e),
+				Err(PacketProcessError::ClientBusy) => sync.write().add_delayed_request(peer, packet_id, data),
+				Ok(()) => {}
+			}
+		}
+	}
+
+	/// Dispatch delayed requests
+	/// The main difference with dispatch packet is the direct send of the responses to the peer
```

Comment:
Should we check if the peer is still connected? Not sure if it's worth it but it seems possible they went away?

Reply:
Added cleanup for disconnected peer